src/modules/Dashboard/components/FlowGrid/FlowGrid.hooks.tsx

```diff
@@ -31,17 +31,15 @@ export function useFlowGridMenu({
   const workspaceRoles = workspaces.useRolesData();
   const roles = appletId ? workspaceRoles?.data?.[appletId] : undefined;
 
-  const canEdit =
-    isManagerOrOwner(roles?.[0]) ||
-    roles?.includes(Roles.Editor) ||
-    roles?.includes(Roles.SuperAdmin);
+  const canEdit = checkIfCanEdit(roles);
 
   const canDoTakeNow =
     featureFlags.enableMultiInformantTakeNow &&
     hasParticipants &&
-    (isManagerOrOwner(roles?.[0]) ||
-      roles?.includes(Roles.Coordinator) ||
-      roles?.includes(Roles.SuperAdmin));
+    (isManagerOrOwner(roles?.[0]) || roles?.includes(Roles.SuperAdmin));
+
+  const canAccessData = checkIfCanAccessData(roles);
```

Comment:
👍 When you messaged me the other day about this, I was already thinking "why aren't we just using a shared util fn here" — I wasn't even aware of `checkIfCanEdit` previously.